onnxruntime/contrib_ops/cuda/bert/tensorrt_fused_multihead_attention/causal/fmha_v2_fp16_Causal_128_32_sm70.cubin.cc

```diff
@@ -17,7 +17,7 @@ namespace onnxruntime {
 namespace contrib {
 namespace cuda {
 
-unsigned char cubin_fmha_v2_fp16_Causal_128_32_sm70_cu_cubin[] = {
+extern const unsigned char cubin_fmha_v2_fp16_Causal_128_32_sm70_cu_cubin[] = {
```

Comment:
It is declared as `extern` in .h file so no need to add extern in .cc file.

This is unconventional because extern typically indicates a declaration without an accompanying definition. Including extern here suggests that the variable might be defined elsewhere, which can lead to confusion and potential linkage issues.

Reply:
> extern typically indicates a declaration without an accompanying definition

In C++ "const" implies "static".  So, if this file didn't correctly include the corresponding header file which contains the declaration, the behavior would be different.

Reply:
I think the "extern" keyword here is optional, either way is fine.  @yuslepukhin , what's your opinion?

Reply:
He probably is on vacation.

Reply:
> I think the "extern" keyword here is optional, either way is fine.

It is not optional here because "const" qualifier without "extern" gives the variable internal linkage. So without it you'll get an "undefined reference" error.

See https://en.cppreference.com/w/cpp/language/cv#Notes

Reply:
No, it is not needed as it is already specified in the header file.

test.h
```c++
#pragma once

extern const int a;
```
test1.cpp:
```c++
#include "test.h"
const int a=1;
```

test2.cpp:
```c++
#include "test.h"
const int a=1;
int main(){}
```
Then  you can use the following commands to build them:
```
g++ -c -o test2.o test2.cpp
g++ -c -o test1.o test1.cpp
gcc -o t test1.o test2.o
```
You will see an error:
```
/usr/bin/ld: test2.o:(.rodata+0x0): multiple definition of `a'; test1.o:(.rodata+0x0): first defined here
collect2: error: ld returned 1 exit status
```
But if you comment out `#include "test.h"`, the error will be gone.

Reply:
> No, it is not needed as it is already specified in the header file.

That doesn't apply here, as *.cc files with FMHA cubins never include fused_multihead_attention_v2.h.